### PR TITLE
Refactor ping-preventing code

### PR DESCRIPTION
**Pull request by Garanas**
This new solution allows you to ping as long as:
 - You are not an observer.
 - Your current focus army is an ally to you.
 - You are not in a replay.

It also ensures that the name of the original author is displayed under the ping. This can be evaded however - in the end we have no full control over the UI code and someone can write a UI mod to prevent this basic check.

 - [x] Disable ping buttons intuitively or communicate that you can not ping when your focus army is not allied to you.

### CODE REVIEW

File: lua/ui/game/gamemain.lua [hunk 1]
```diff
@@ -136,6 +136,11 @@ function OnFirstUpdate()
 end
 
 function CreateUI(isReplay)
+
+    -- TODO: is this the best place? I tried that userInit.lua but because of the preloading that starts loading during the lobby and it crashes.
```
Comment (Garanas) [inline lua/ui/game/gamemain.lua:140]: Please make suggestions on this todo.

**Comment by KionX** [comment]
It is better to make a revert for #3360 and #3240

**Comment by Garanas** [comment]
As in, revert through Git?

I have no experience with that @shalkya could you take care of that?

**Comment by Garanas** [comment]
Woops - did something wrong.

**Comment by Garanas** [comment]
Ready to be squashed 👍

**Comment by KionX** [comment]
Tested it. Works.

I have an idea.
After death, do not automatically switch to the observer.
Prohibit ping if you were an observer.

**Pull request merged**